contracts/pcv/maker/MakerRouter.sol

```diff
@@ -0,0 +1,229 @@
+        address to
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        _redeemFromFeiPSM(amountFeiIn, minDaiAmountOut, address(this));
+        daiPSM.buyGem(to, (minDaiAmountOut) / USDC_SCALING_FACTOR);
+    }
+
+    /// @notice Function to swap all of FEI balance to USDC
+    /// @param to the address the USDC should be sent to once swapped
+    function swapAllFeiForUsdc(address to)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        uint256 minDaiAmountOut = _redeemAllBalanceFromFeiPSM(address(this));
+        daiPSM.buyGem(to, (minDaiAmountOut) / USDC_SCALING_FACTOR);
+    }
+
+    /// @notice Function to swap for both DAI and USDC
+    /// @dev Function will swap from FEI to DAI first then DAI to USDC
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+    /// @param usdcTo the address the USDC should be sent to once swapped
+    /// @param daiTo the address the DAI should be sent to once swapped
+    function swapFeiForUsdcAndDai(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        address usdcTo,
+        address daiTo,
+        uint256 ratioUSDC
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        require(
+            ratioUSDC < Constants.BASIS_POINTS_GRANULARITY && ratioUSDC > 0,
+            "MakerRouter: Invalid USDC Ratio"
+        );
+
+        _redeemFromFeiPSM(amountFeiIn, minDaiAmountOut, address(this));
+        _swapForUsdcAndDai(minDaiAmountOut, usdcTo, daiTo, ratioUSDC);
+    }
+
+    /// @notice Function to swap all FEI balance for both DAI and USDC
+    /// @param usdcTo the address the USDC should be sent to once swapped
+    /// @param daiTo the address the DAI  should be sent to once swapped
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+    function swapAllFeiForUsdcAndDai(
+        address usdcTo,
+        address daiTo,
+        uint256 ratioUSDC
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        uint256 minDaiAmountOut = _redeemAllBalanceFromFeiPSM(address(this));
+        _swapForUsdcAndDai(minDaiAmountOut, usdcTo, daiTo, ratioUSDC);
+    }
+
+    /// @notice Function to withdraw tokens to an address
+    /// @param token the token to withdraw
+    /// @param amount the amount to send
+    /// @param to the address the token should be sent to
+    function withdrawERC20(
+        address token,
+        uint256 amount,
+        address to
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        IERC20(token).safeTransfer(to, amount);
```

Comment:
Would definitely emit an event here to make the use of this function more trackable